### PR TITLE
Add All Controls benchmark

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: controls_by_service/search.sp
Comment (madhushreeray30) [inline controls_by_service/search.sp:1]: Should we rename this file to `cognitivesearch.sp`?

File: regulatory_compliance/appservice.sp [hunk 1]
```diff
@@ -284,129 +276,135 @@ control "appservice_function_app_latest_java_version" {
...
+control "appservice_web_app_incoming_client_cert_on" {
+  title       = "Ensure the web app has 'Client Certificates (Incoming client certificates)' set to 'On'"
+  description = "Client certificates allow for the app to request a certificate for incoming requests. Only clients that have a valid certificate will be able to reach the app."
+  query       = query.appservice_web_app_incoming_client_cert_on
+
+  tags = local.regulatory_compliance_appservice_common_tags
+}
+
+control "appservice_authentication_enabled" {
+  title       = "Ensure App Service Authentication is set up for apps in Azure App Service"
```
Comment (madhushreeray30) [inline regulatory_compliance/appservice.sp:387]:
```suggestion
  title       = "Ensure App Service authentication is set up for apps in Azure App Service"
```

File: regulatory_compliance/appservice.sp [hunk 2]
```diff
@@ -284,129 +276,135 @@ control "appservice_function_app_latest_java_version" {
...
+  description = "Azure App Service Authentication is a feature that can prevent anonymous HTTP requests from reaching a Web Application or authenticate those with tokens before they reach the app. If an anonymous request is received from a browser, App Service will redirect to a logon page. To handle the logon process, a choice from a set of identity providers can be made, or a custom authentication mechanism can be implemented."
```
Comment (madhushreeray30) [inline regulatory_compliance/appservice.sp:388]:
```suggestion
  description = "Azure App Service authentication is a feature that can prevent anonymous HTTP requests from reaching a Web Application or authenticate those with tokens before they reach the app. If an anonymous request is received from a browser, App Service will redirect to a logon page. To handle the logon process, a choice from a set of identity providers can be made, or a custom authentication mechanism can be implemented."
```

File: regulatory_compliance/iam.sp [hunk 3]
```diff
@@ -135,6 +135,39 @@ control "iam_user_with_owner_permission_on_subscription_mfa_enabled" {
   })
 }
 
+control "iam_no_custom_subscription_owner_roles_created" {
+  title       = "Ensure That No Custom Subscription Administrator Roles Exist"
```
Comment (madhushreeray30) [inline regulatory_compliance/iam.sp:139]:
```suggestion
  title       = "Ensure that no Custom Subscription Administrator roles exist"
```

File: regulatory_compliance/iam.sp [hunk 4]
```diff
@@ -135,6 +135,39 @@ control "iam_user_with_owner_permission_on_subscription_mfa_enabled" {
...
+  description = "The principle of least privilege should be followed and only necessary privileges should be assigned instead of allowing full administrative access."
+  query       = query.iam_no_custom_subscription_owner_roles_created
+
+  tags = local.regulatory_compliance_iam_common_tags
+}
+
+control "iam_conditional_access_mfa_enabled" {
+  title       = "Ensure Multi-factor Authentication is Required for Azure Management"
```
Comment (madhushreeray30) [inline regulatory_compliance/iam.sp:147]:
```suggestion
  title       = "Ensure Multi-factor Authentication is required for Azure Management"
```

File: regulatory_compliance/iam.sp [hunk 5]
```diff
@@ -135,6 +135,39 @@ control "iam_user_with_owner_permission_on_subscription_mfa_enabled" {
...
+control "iam_user_not_allowed_to_create_security_group" {
+  title       = "Ensure that 'Users can create security groups in Azure portals, API or PowerShell' is set to 'No'"
+  description = "Restrict security group creation to administrators only."
+  query       = query.iam_user_not_allowed_to_create_security_group
+
+  tags = local.regulatory_compliance_iam_common_tags
+}
+
+control "iam_user_not_allowed_to_register_application" {
+  title       = "Ensure That 'Users Can Register Applications' Is Set to 'No'"
```
Comment (madhushreeray30) [inline regulatory_compliance/iam.sp:164]:
```suggestion
  title       = "Ensure that 'Users Can Register Applications' is set to 'No'"
```

File: regulatory_compliance/monitor.sp [hunk 6]
```diff
@@ -80,9 +80,151 @@ control "monitor_log_profile_retention_365_days" {
...
+  title       = "Ensure the storage account containing the container with activity logs is encrypted with Customer Managed Key"
+  description = "Storage accounts with the activity log exports can be configured to use Customer Managed Keys (CMK)."
+  query       = query.monitor_logs_storage_container_insights_operational_logs_encrypted_with_byok
+
+  tags = local.regulatory_compliance_monitor_common_tags
+}
+
+control "monitor_logs_storage_container_insights_operational_logs_not_public_accessible" {
+  title       = "Ensure the storage container storing the activity logs is not publicly accessible"
+  description = "The storage account container containing the activity log export should not be publicly accessible."
```
Comment (madhushreeray30) [inline regulatory_compliance/monitor.sp:216]:
```suggestion
  title       = "Ensure the storage container storing the operational logs is not publicly accessible"
  description = "The storage account container containing the operational log export should not be publicly accessible."
```

File: regulatory_compliance/mysql.sp [hunk 7]
```diff
@@ -88,6 +88,30 @@ control "mysql_server_encrypted_at_rest_using_cmk" {
   })
 }
 
+control "mysql_server_audit_logging_enabled" {
+  title       = "Ensure server parameter 'audit_log_enabled' is set to 'ON' for MySQL Database Server"
+  description = "Enable audit_log_enabled on MySQL Servers."
```
Comment (madhushreeray30) [inline regulatory_compliance/mysql.sp:93]:
```suggestion
  description = "Enable audit logging on MySQL Servers."
```

File: regulatory_compliance/network.sp [hunk 8]
```diff
@@ -137,6 +131,47 @@ control "network_subnet_protected_by_firewall" {
   })
 }
 
+control "network_bastion_host_min_1" {
+  title       = "Ensure an Azure Bastion Host Exists"
```
Comment (madhushreeray30) [inline regulatory_compliance/network.sp:135]:
```suggestion
  title       = "Ensure an Azure Bastion Host exists"
```

File: regulatory_compliance/securitycenter.sp [hunk 9]
```diff
@@ -160,9 +156,63 @@ control "securitycenter_pricing_standard" {
...
+control "securitycenter_asc_default_setting_not_disabled" {
+  title       = "Ensure any of the ASC Default policy setting is not set to \"Disabled\""
+  description = "None of the settings offered by ASC Default policy should be set to effect \"Disabled\"."
+  query       = query.securitycenter_asc_default_setting_not_disabled
+
+  tags = local.regulatory_compliance_securitycenter_common_tags
+}
+
+control "securitycenter_azure_defender_on_for_cosmosdb" {
+  title       = "Ensure That Microsoft Defender for Azure Cosmos DB Is Set To 'On'"
```
Comment (madhushreeray30) [inline regulatory_compliance/securitycenter.sp:179]:
```suggestion
  title       = "Ensure That Microsoft Defender for Azure Cosmos DB is set to 'On'"
```

File: regulatory_compliance/securitycenter.sp [hunk 10]
```diff
@@ -160,9 +156,63 @@ control "securitycenter_pricing_standard" {
...
+  description = "Microsoft Defender for Azure Cosmos DB scans all incoming network requests for threats to your Azure Cosmos DB resources."
+  query       = query.securitycenter_azure_defender_on_for_cosmosdb
+
+  tags = local.regulatory_compliance_securitycenter_common_tags
+}
+
+control "securitycenter_azure_defender_on_for_database" {
+  title       = "Ensure That Microsoft Defender for Databases Is Set To 'On'"
```
Comment (madhushreeray30) [inline regulatory_compliance/securitycenter.sp:187]:
```suggestion
  title       = "Ensure That Microsoft Defender for Databases is set to 'On'"
```

File: regulatory_compliance/securitycenter.sp [hunk 11]
```diff
@@ -160,9 +156,63 @@ control "securitycenter_pricing_standard" {
...
+  description = "Turning on Microsoft Defender for Databases enables threat detection for the instances running your database software. This provides threat intelligence, anomaly detection, and behavior analytics in the Azure Microsoft Defender for Cloud. Instead of being enabled on services like Platform as a Service (PaaS), this implementation will run within your instances as Infrastructure as a Service (IaaS) on the Operating Systems hosting your databases."
+  query       = query.securitycenter_azure_defender_on_for_database
+
+  tags = local.regulatory_compliance_securitycenter_common_tags
+}
+
+control "securitycenter_azure_defender_on_for_opensource_relational_db" {
+  title       = "Ensure That Microsoft Defender for Open-Source Relational Databases Is Set To 'On'"
```
Comment (madhushreeray30) [inline regulatory_compliance/securitycenter.sp:195]:
```suggestion
  title       = "Ensure That Microsoft Defender for Open-Source Relational Databases is set to 'On'"
```

File: regulatory_compliance/securitycenter.sp [hunk 12]
```diff
@@ -160,9 +156,63 @@ control "securitycenter_pricing_standard" {
...
+  description = "Turning on Microsoft Defender for Open-source relational databases enables threat detection for Open-source relational databases, providing threat intelligence, anomaly detection, and behavior analytics in the Microsoft Defender for Cloud."
+  query       = query.securitycenter_azure_defender_on_for_opensource_relational_db
+
+  tags = local.regulatory_compliance_securitycenter_common_tags
+}
+
+control "securitycenter_mcas_integration" {
+  title       = "Ensure that Microsoft Defender for Cloud Apps (MCAS) Integration with Microsoft Defender for Cloud is Selected"
```
Comment (madhushreeray30) [inline regulatory_compliance/securitycenter.sp:203]:
```suggestion
  title       = "Ensure that Microsoft Defender for Cloud Apps (MCAS) Integration with Microsoft Defender for Cloud is selected"
```

File: regulatory_compliance/storage.sp [hunk 13]
```diff
@@ -120,6 +120,62 @@ control "storage_account_encryption_scopes_encrypted_at_rest_with_cmk" {
...
+  title       = "Ensure that 'Public access level' is set to Private for blob containers"
+  description = "Disable anonymous access to blob containers and disallow blob public access on storage account."
+  query       = query.storage_account_blob_containers_public_access_private
+
+  tags = local.regulatory_compliance_storage_common_tags
+}
+
+control "storage_account_blob_service_logging_enabled" {
+  title       = "Ensure Storage logging is enabled for Blob service for read, write, and delete requests"
+  description = "The Storage Blob service provides scalable, cost-efficient objective storage in the cloud. Storage Logging happens server-side and allows details for both successful and failed requests to be recorded in the storage account. These logs allow users to see the details of read, write, and delete operations against the blobs. Storage Logging log entries contain the following information about individual requests: Timing information such as start time, end-to-end latency, and server latency, authentication details , concurrency information and the sizes of the request and response messages."
```
Comment (madhushreeray30) [inline regulatory_compliance/storage.sp:133]:
```suggestion
  description = "The Storage Blob service provides scalable, cost-efficient objective storage in the cloud. Storage Logging happens server-side and allows details for both successful and failed requests to be recorded in the storage account. These logs allow users to see the details of read, write, and delete operations against the blobs. Storage Logging log entries contain the following information about individual requests: Timing information such as start time, end-to-end latency, and server latency, authentication details, concurrency information, and the sizes of the request and response messages."
```

File: regulatory_compliance/storage.sp [hunk 14]
```diff
@@ -120,6 +120,62 @@ control "storage_account_encryption_scopes_encrypted_at_rest_with_cmk" {
...
+  query       = query.storage_account_blob_service_logging_enabled
+
+  tags = local.regulatory_compliance_storage_common_tags
+}
+
+control "storage_account_table_service_logging_enabled" {
+  title       = "Ensure Storage Logging is Enabled for Table Service for 'Read', 'Write', and 'Delete' Requests "
```
Comment (madhushreeray30) [inline regulatory_compliance/storage.sp:140]:
```suggestion
  title       = "Ensure Storage Logging is Enabled for Table Service for 'Read', 'Write', and 'Delete' requests"
```

File: regulatory_compliance/storage.sp [hunk 15]
```diff
@@ -120,6 +120,62 @@ control "storage_account_encryption_scopes_encrypted_at_rest_with_cmk" {
...
+  query       = query.storage_account_min_tls_1_2
+
+  tags = local.regulatory_compliance_storage_common_tags
+}
+
+control "storage_account_queue_services_logging_enabled" {
+  title       = "Ensure Storage logging is enabled for Queue service for read, write, and delete requests"
+  description = "The Storage Queue service stores messages that may be read by any client who has access to the storage account. A queue can contain an unlimited number of messages, each of which can be up to 64KB in size using version 2011-08-18 or newer. Storage Logging happens server-side and allows details for both successful and failed requests to be recorded in the storage account. These logs allow users to see the details of read, write, and delete operations against the queues. Storage Logging log entries contain the following information about individual requests: Timing information such as start time, end-to-end latency, and server latency, authentication details , concurrency information and the sizes of the request and response messages."
```
Comment (madhushreeray30) [inline regulatory_compliance/storage.sp:157]:
```suggestion
  description = "The Storage Queue service stores messages that may be read by any client who has access to the storage account. A queue can contain an unlimited number of messages, each of which can be up to 64KB in size using version 2011-08-18 or newer. Storage Logging happens server-side and allows details for both successful and failed requests to be recorded in the storage account. These logs allow users to see the details of read, write, and delete operations against the queues. Storage Logging log entries contain the following information about individual requests: Timing information such as start time, end-to-end latency, and server latency, authentication details, concurrency information, and the sizes of the request and response messages."
```

**Review by madhushreeray30 — COMMENTED** [review]
@khushboo9024 please have a look at the comments, thanks!

File: controls_by_service/controls_by_service.sp [hunk 16]
```diff
@@ -0,0 +1,57 @@
+locals {
+  controls_by_service_common_tags = merge(local.azure_compliance_common_tags, {
+    type = "Benchmark"
+  })
+}
+benchmark "controls_by_service" {
+  title       = "Controls by Service"
+  description = "This benchmark contains all controls grouped by service to help you detect resource configurations that do not meet best practices."
```
Comment (cbruno10) [inline controls_by_service/controls_by_service.sp:8]:
```suggestion
benchmark "all_controls" {
  title       = "All Controls"
  description = "This benchmark contains all controls grouped by service to help you detect resource configurations that do not meet best practices."
```

File: controls_by_service/controls_by_service.sp [hunk 17]
```diff
@@ -0,0 +1,57 @@
+locals {
+  controls_by_service_common_tags = merge(local.azure_compliance_common_tags, {
```
Comment (cbruno10) [inline controls_by_service/controls_by_service.sp:2]:
```suggestion
  all_controls_common_tags = merge(local.azure_compliance_common_tags, {
```

File: controls_by_service/controls_by_service.sp [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+    type = "Benchmark"
+  })
+}
+benchmark "controls_by_service" {
+  title       = "Controls by Service"
+  description = "This benchmark contains all controls grouped by service to help you detect resource configurations that do not meet best practices."
+  children = [
+    benchmark.controls_by_service_activedirectory,
```
Comment (cbruno10) [inline controls_by_service/controls_by_service.sp:10]:
```suggestion
    benchmark.all_controls_activedirectory,
```

Please make the changes for other benchmark names too, thanks!

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@khushboo9024 Please see comments, thanks!

**Review by madhushreeray30 — APPROVED** [review]

**Review by misraved — APPROVED** [review]

**Pull request merged**